### PR TITLE
Feature: API added and minor edits to overall code

### LINKED ISSUE #5: Api's

### PR DESCRIPTION
**Pull request by zsh28**
- More detailed form to add flight details
- Added close form button
- Edits in formatting made
- Added APIs: Flight and Weather

### CODE REVIEW

File: Screens/Home.tsx [hunk 1]
```diff
@@ -1,58 +1,168 @@
...
-  const openModal =() => setModalVisible(true);
-  //set modal to visible -> False
-  const closeModal =() => setModalVisible(false);
+  const togglePicker = () => {
+    setPickerVisible(!isPickerVisible);
+  };
 
-  return <View style={styles.container}>
-    <Ionicons
+  const ToggleModal = () => {
```
Comment (SwyftPain) [inline Screens/Home.tsx:29]: Inconsistent casing. JavaScript/TypeScript mainly uses camel-case. Meaning first letter is lowercase.
Your togglePicker function follows this standard. ToggleModal doesn't. Not a big deal and can be fixed in a later formatting/consistency PR but just a note to remember :)

File: Screens/Home.tsx [hunk 2]
```diff
@@ -1,58 +1,168 @@
...
+    }
+
+    try {
+      const flightData = await getFlightData(
+        flightForm.departure,
+        flightForm.arrival,
+        date,
+        flightForm.flightNumber
+      );
+      console.log("Flight Data Received:", flightData);
```
Comment (SwyftPain) [inline Screens/Home.tsx:63]: These logs are probably not needed beyond/after testing and actual implementation. Once the app is built, you won't be able to see them anywhere anyway.

File: Screens/Home.tsx [hunk 3]
```diff
@@ -1,58 +1,168 @@
...
               style={styles.input}
-              placeholder="Flight Number"
+              placeholder="Arrival e.g. JFK"
               placeholderTextColor={styles.placeholder.color}
+              onChangeText={(text) => handleInputChange("arrival", text)}
+              value={flightForm.arrival}
             />
-            <GreenButton
-              onPress={closeModal}
+            <TouchableOpacity onPress={togglePicker}>
```
Comment (SwyftPain) [inline Screens/Home.tsx]: We already made a Button component, we could have just used that. If it didn't have needed props on it, just add them.

File: Screens/Settings.tsx [hunk 4]
```diff
@@ -10,48 +10,51 @@ const Settings = () => {
...
+        placeholderTextColor={styles.placeholder.color}
+      />
+      <Button
+        onPress={changeEmail}
+        disabled={false}
+        variant={"green"}
+        text={"Change Email"}
+      />
+      <View style={{ height: 20 }}></View>
       <Header headerTitle={"Change Password"} />
```
Comment (SwyftPain) [inline Screens/Settings.tsx:39]: Slightly confused, do we have two headers in the same file?

File: Styles/DarkStyles.tsx [hunk 5]
```diff
@@ -57,6 +48,15 @@ const styles = StyleSheet.create({
     bottom: 20,
     right: 20,
   },
+  //close icon
```
Comment (SwyftPain) [inline Styles/DarkStyles.tsx:51]: isn't this obvious its close icon?

File: api/weather.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import axios from "axios";
+/*
+const options = {
+  method: 'GET',
+  url: 'https://weatherapi-com.p.rapidapi.com/current.json',
+  params: {q: 'London'},
+  headers: {
+    'X-RapidAPI-Key': '5bda5892bfmsh0930416066adb65p16c0bcjsnf58b3e7581bf',
```
Comment (SwyftPain) [inline api/weather.tsx]: Do we want this info in our code? (Benefit of the doubt, could be from documentation example)

File: api/weather.tsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+const getWeatherData = async (location: String) => {
+  const options = {
+    method: "GET",
+    url: "https://weatherapi-com.p.rapidapi.com/current.json",
+    params: { q: location },
+    headers: {
+      "X-RapidAPI-Key": process.env.EXPO_PUBLIC_API_KEY,
+      "X-RapidAPI-Host": "weatherapi-com.p.rapidapi.com",
+    },
+  };
```
Comment (SwyftPain) [inline api/weather.tsx:11]: Formatting inconsistency, new lines, etc

File: package.json [hunk 8]
```diff
@@ -20,15 +20,20 @@
     "expo": "50.0.14",
     "expo-dev-client": "3.3.11",
     "expo-status-bar": "1.11.1",
+    "fast-xml-parser": "^4.3.6",
```
Comment (SwyftPain) [inline package.json:23]: The ^ symbol should be removed

File: package.json [hunk 9]
```diff
@@ -20,15 +20,20 @@
...
     "react": "18.2.0",
     "react-native": "0.73.6",
     "react-native-gesture-handler": "2.14.0",
+    "react-native-modern-datepicker": "^1.0.0-beta.91",
```
Comment (SwyftPain) [inline package.json]: The ^ symbol should be removed

File: package.json [hunk 10]
```diff
@@ -20,15 +20,20 @@
...
     "react-native-safe-area-context": "4.9.0",
     "react-native-screens": "3.30.1"
   },
   "devDependencies": {
     "@babel/core": "7.20.2",
     "@types/react": "18.2.45",
+    "@types/react-native-modern-datepicker": "^1.0.5",
```
Comment (SwyftPain) [inline package.json]: The ^ symbol should be removed

**Review by SwyftPain — CHANGES_REQUESTED** [review]
Some minor inconsistencies which are fine. The versions in package.json need to be edited.

Some abstractions or components could be used like <Button /> rather than recreating it again.

Also with changes like putting 2 headers in a file, if not screenshots, would be at least nice to have a message/comment to know what exactly was being added with the PR, since its not just creating API functions, it is actually adding/editing other files as well.

So in that sense, just "API Added" is not clear enough description.

Possibly better solution would be to provide bulletpoints, saying what files were edited, how and why in a short sentence.

**Review by SwyftPain — APPROVED** [review]
Seems good now, i appreciate some consistency changes.

**Review by avrey-s — APPROVED** [review]

**Pull request merged**